pkg/scheduler/plugins/predicates/predicates.go

```diff
@@ -432,31 +432,34 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 		if predicate.cacheEnable {
 			fit, err = pCache.PredicateWithCache(node.Name, task.Pod)
 			if err != nil {
-				predicateCacheStatus, fit, err = predicateByStablefilter(task.Pod, nodeInfo)
+				predicateCacheStatus, fit, _ = predicateByStablefilter(task.Pod, nodeInfo)
 				pCache.UpdateCache(node.Name, task.Pod, fit)
 			} else {
 				if !fit {
 					err = fmt.Errorf("plugin equivalence cache predicates failed")
+					predicateCacheStatus = append(predicateCacheStatus, &api.Status{
+						Code: api.Error, Reason: err.Error(), Plugin: CachePredicate,
+					})
 				}
 			}
 		} else {
-			predicateCacheStatus, fit, err = predicateByStablefilter(task.Pod, nodeInfo)
+			predicateCacheStatus, fit, _ = predicateByStablefilter(task.Pod, nodeInfo)
 		}
 
 		predicateStatus = append(predicateStatus, predicateCacheStatus...)
 		if !fit {
-			return predicateStatus, err
+			return api.NewFitErrWithStatus(task, node, predicateStatus...)
 		}
 
 		// Check NodePort
 		if predicate.nodePortEnable {
 			isSkipNodePorts := handleSkipPredicatePlugin(task, skipPlugins, nodePortFilter.Name(), node)
 			if !isSkipNodePorts {
 				status := nodePortFilter.Filter(context.TODO(), state, nil, nodeInfo)
-				nodePortStatus := framework.ConvertPredicateStatus(status)
+				nodePortStatus := api.ConvertPredicateStatus(status)
 				if nodePortStatus.Code != api.Success {
 					predicateStatus = append(predicateStatus, nodePortStatus)
-					return predicateStatus, fmt.Errorf("plugin %s predicates failed %s", nodePortFilter.Name(), status.Message())
+					return api.NewFitErrWithStatus(task, node, predicateStatus...)
```

Comment:
When multiple filtering algorithms are executed in the predicate plug-in, for example, nodePortFilter, podAffinityFilter, volumeZoneFilter, and podTopologySpreadFilter,

If the result of nodePortFilter is Unschedulable, the result of podAffinityFilter is UnschedulableAndUnresolvable. The predicate plug-in is returned after nodePortFilter is executed. Other algorithm plug-ins are not executed.

When the preempt action is used to determine whether the current node meets the conditions, the corresponding status code contains only the nodePortFilter information and is Unschedulable, but does not contain the UnschedulableAndUnresolvable information of podAffinityFilter. In this case, the system considers that the node meets the preemption conditions, resulting in misjudgment.

Of course, this is an existing problem, not introduced by this pr, but we can discuss the idea together.

Reply:
That is a problem in current architecture.  We have discussed this problem with @Monokaix and we can following the kubernetes process. And the preemtion/reclaim action can also use the predicating result cache.

I think we can record a issue under https://github.com/volcano-sh/volcano/issues/3502.